src/home/sorting/sorting-manager.ts

```diff
@@ -50,22 +54,36 @@ export class SortingManager {
     textBox.value = searchQuery;
 
     const issuesContainer = document.getElementById("issues-container") as HTMLDivElement;
-
-    function filterIssues() {
+    const filterIssues = () => {
       try {
         const filterText = textBox.value.toLowerCase();
         const issues = Array.from(issuesContainer.children) as HTMLDivElement[];
-        issues.forEach((issue) => {
-          const issueId = issue.children[0].getAttribute("data-issue-id");
-          issue.classList.add("active");
-          if (!issueId) return;
-          const gitHubIssue = taskManager.getGitHubIssueById(parseInt(issueId));
-          if (!gitHubIssue) return;
-          const searchableProperties = ["title", "body", "number", "html_url"] as const;
-          const searchableStrings = searchableProperties.map((prop) => gitHubIssue[prop]?.toString().toLowerCase());
-          const isVisible = searchableStrings.some((str) => str?.includes(filterText));
-          issue.style.display = isVisible ? "block" : "none";
-        });
+        // Get issue IDs and search results
+        const issueIds = issues
+          .map((issue) => issue.children[0].getAttribute("data-issue-id"))
+          .filter((id): id is string => id !== null)
+          .map((id) => parseInt(id));
+
+        const searchResults = taskManager.issueSearcher.search(filterText, issueIds);
+        // If there's a search term, sort by relevance
+        if (filterText) {
+          issues
+            .sort((a, b) => {
+              const scoreA = parseFloat(a.getAttribute("data-relevance-score") || "0");
+              const scoreB = parseFloat(b.getAttribute("data-relevance-score") || "0");
+              return scoreB - scoreA; // Sort in descending order of relevance score
+            })
+            .forEach((issue) => {
+              const issueId = issue.children[0].getAttribute("data-issue-id");
+              if (!issueId) return;
```

Comment:
Just add `issue.classList.add("active");` above `if (!issueId) return;` to fix that loading problem